crates/pink/sidevm/host-runtime/src/env.rs

```diff
@@ -213,18 +213,27 @@ impl Env {
         reply_tx: OneshotSender<Vec<u8>>,
     ) -> Option<impl Future<Output = anyhow::Result<()>>> {
         let mut env_guard = self.inner.lock().unwrap();
+        let tx = env_guard.state.query_tx.clone()?;
         let reply_tx = env_guard
             .state
             .resources
             .push(Resource::OneshotTx(Some(reply_tx)));
-        let tx = env_guard.state.query_tx.clone()?;
+        let inner = self.inner.clone();
         Some(async move {
+            let reply_tx = reply_tx?;
             let query = QueryRequest {
                 origin,
                 payload,
-                reply_tx: reply_tx?,
+                reply_tx,
             };
-            tx.send(query.encode()).await?;
+            let result = tx.send(query.encode()).await;
+            if result.is_err() {
```

Comment:
Most of the case `result` could be error if the queue is full of unprocessed messages I guess? But we cannot really do much besides setting a timeout for reading the response. But at least, this doesn't cost us a lot. Is my understanding correct?

Reply:
Exactly. It doesn't cost us a lot, at most the channel cap number of fds. It happens when the guest is stuck somewhere. Not a big problem.

Reply:
Cool. I'm good with the fix